### PR TITLE
update passwordSetByUser flag when reset or change password

### PR DESCRIPTION
**Pull request by osiievska23**

### CODE REVIEW

**Review by amedvedchuk — APPROVED** [review]

File: src/main/java/com/icthh/xm/uaa/domain/UserLoginType.java [hunk 1]
```diff
@@ -51,15 +52,15 @@ public static UserLoginType fromString(String value) {
 
     public static UserLoginType getByRegex(String value) {
         if (value == null || value.isEmpty()) {
-            throw new IllegalArgumentException("can not build UserLoginType from value: " + value);
+            throw new BusinessException("User login type could not be determined by value: " + value);
```
Comment (sergeysenja1992) [inline src/main/java/com/icthh/xm/uaa/domain/UserLoginType.java:55]: add error code

File: src/main/java/com/icthh/xm/uaa/domain/UserLoginType.java [hunk 2]
```diff
@@ -51,15 +52,15 @@ public static UserLoginType fromString(String value) {
...
         }
         if (EmailValidator.getInstance().isValid(value)) {
             return UserLoginType.EMAIL;
         }
         if (PHONE_PATTERN.matcher(value).matches()) {
             return UserLoginType.MSISDN;
         }
-        throw new IllegalArgumentException("can not build UserLoginType from value: " + value);
+        throw new BusinessException("User login type could not be determined by value: " + value);
```
Comment (sergeysenja1992) [inline src/main/java/com/icthh/xm/uaa/domain/UserLoginType.java:63]: add error code

**Review by sergeysenja1992 — APPROVED** [review]

File: src/main/java/com/icthh/xm/uaa/domain/UserLoginType.java (hunk 1 above)
Comment (osiievska23) [reply src/main/java/com/icthh/xm/uaa/domain/UserLoginType.java]: done

File: src/main/java/com/icthh/xm/uaa/domain/UserLoginType.java (hunk 2 above)
Comment (osiievska23) [reply src/main/java/com/icthh/xm/uaa/domain/UserLoginType.java]: done

**Pull request merged**